Source/Dafny/Compiler-go.cs

```diff
@@ -1831,9 +1831,27 @@ private class SpecialNativeType : UserDefinedType {
     }
 
     protected override void EmitPrintStmt(TargetWriter wr, Expression arg) {
-      wr.Write("_dafny.Print(");
-      TrExpr(arg, wr, false);
-      wr.WriteLine(")");
+      bool isString = arg.Type.AsCollectionType != null &&
+                      arg.Type.AsCollectionType.AsSeqType != null &&
+                      arg.Type.AsCollectionType.AsSeqType.Arg.IsCharType;
```

Comment:
`arg.Type.AsCollectionType.AsSeqType` gives the same result as just `arg.Type.AsSeqType`.

Same comment applies several places below.

Reply:
Done